oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -746,67 +746,143 @@ impl wasmi::ModuleImportResolver for WasmInterface {
     }
 }
 
-/// Create a new instance of a Wasm node. If the entry point is not found,
-/// `ERR(OakStatus::ErrInvalidArgs)` will be returned immediately.
-pub fn new_instance(
-    config_name: &str,
+pub struct WasmNode {
```

Comment:
@blaxill I wonder if this should just be merged with `WasmInterface`?

Reply:
Currently the WasmNode is not passed to the executed thread which might complicate resolving this, perhaps the approach I outlined below would help with this?